app/models/batch_connection.rb

```diff
@@ -3,4 +3,17 @@
 class BatchConnection < ApplicationRecord
   belongs_to :batch_process
   belongs_to :connectable, polymorphic: true
+
+  def child_object_count
```

Comment:
I think this method should exist only on the parent object it never needs to exist in this class. 

```ruby
batch_connection.connectable&.child_object_count
```